File: src/accessibility_manager.cpp
#include <string.h>
#include <iostream>

#include "accessibility_manager.h"
#include "libnativeapi/include/nativeapi.h"

using namespace nativeapi;

static AccessibilityManager g_accessibility_manager = AccessibilityManager();

FFI_PLUGIN_EXPORT
void accessibility_manager_enable() {
  g_accessibility_manager.Enable();
}

FFI_PLUGIN_EXPORT
bool accessibility_manager_is_enabled() {
  return g_accessibility_manager.IsEnabled();
}

File: packages/cnativeapi/lib/src/accessibility_manager.dart
import 'package:nativeapi/src/ffi/bindings_generated.dart';
import 'ffi/bindings.dart';

class AccessibilityManager {
  AccessibilityManager._();

  static final AccessibilityManager instance = AccessibilityManager._();

  NativeApiBindings get _bindings => nativeApiBindings;

  /// Enable the accessibility permission.
  void enable() {
    _bindings.native_accessibility_manager_enable();
  }

  /// Whether the accessibility permission is enabled.
  bool isEnabled() {
    return _bindings.native_accessibility_manager_is_enabled();
  }
}
